### PR TITLE
feat: support stack for haskell

### PR DESCRIPTION
**Pull request by luochen1990**
stack have been the most commonly used tool for haskell, this PR add stack support for haskell

### CODE REVIEW

File: doc/SingleCompile.txt [hunk 1]
```diff
@@ -744,7 +744,9 @@ Go:
     go (Go)
 
 Haskell:
+    stack-ghc (The Haskell Tool Stack (stack exec -- ghc))
     ghc (Glasgow Haskell Compiler)
+    stack-runhaskell (The Haskell Tool Stack (stack exec -- runhaskell))
```
Comment (xuhdev) [inline doc/SingleCompile.txt]: Can you follow an alphabetical order? The list strictly follows that.

File: autoload/SingleCompile/templates/haskell.vim [hunk 2]
```diff
@@ -18,12 +18,28 @@
 " check doc/SingleCompile.txt for more information
 
 function! SingleCompile#templates#haskell#Initialize()
+    " stack-runhaskell
+    call SingleCompile#SetCompilerTemplate('haskell', 'stack-runhaskell',
+               \'The Haskell Tool Stack (stack exec -- runhaskell)', 'stack', 'exec -- runhaskell', '')
+    call SingleCompile#SetPriority('haskell', 'stack-runhaskell', 10)
```
Comment (xuhdev) [inline autoload/SingleCompile/templates/haskell.vim]: Can you explain why you give "stack" series a higher priority? I haven't touched haskell for a long time and I couldn't find a comparison. Is it more popular now?

Comment (luochen1990) [reply autoload/SingleCompile/templates/haskell.vim]: Yes, stack is more popular, and it can manager different version of GHC automatically.

**Comment by xuhdev** [comment]
Thanks!

**Pull request merged**